Remove z_const workaround for WinCairo in third_party/zlib/google/compression_utils_portable.cc

fujii opened the pull request:
#### bd274655c406555a1422bc01f50407109e94b9ad
<pre>
Remove z_const workaround for WinCairo in third_party/zlib/google/compression_utils_portable.cc
<a href="https://bugs.webkit.org/show_bug.cgi?id=276515">https://bugs.webkit.org/show_bug.cgi?id=276515</a>

Reviewed by Don Olmstead.

&lt;<a href="https://commits.webkit.org/227764@main">https://commits.webkit.org/227764@main</a>&gt; added a workaround for zlib-ng.
zlib-ng fixed the problem. &lt;<a href="https://github.com/zlib-ng/zlib-ng/pull/704">https://github.com/zlib-ng/zlib-ng/pull/704</a>&gt;

* Source/ThirdParty/ANGLE/third_party/zlib/google/compression_utils_portable.cc:

Canonical link: <a href="https://commits.webkit.org/280930@main">https://commits.webkit.org/280930@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/34e5a5b44caafcea0b46ee8f33602daedf34e421

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/57934 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/37262 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/10410 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/61556 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/8379 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/44898 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/8567 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/46951 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/5967 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/59964 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/34945 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/50075 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/27777 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/31711 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/7369 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/7383 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/53659 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/7638 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/63243 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/1848 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/7707 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/54176 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/1855 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/50086 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/54311 "Found 2 new API test failures: /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/text/state-changed, /WebKitGTK/TestInspectorServer:/webkit/WebKitWebInspectorServer/http-test-page-list (failure)") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/1578 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8662 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/33091 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/34177 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/35261 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/33922 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

donny-dont reviewed (approved):
This is fine but it doesn't feel like this is synced with the one in the repository

https://github.com/chromium/chromium/blob/main/third_party/zlib/google/compression_utils_portable.cc

There something up with an ANGLE roll here?

fujii committed:
Remove z_const workaround for WinCairo in third_party/zlib/google/compression_utils_portable.cc
https://bugs.webkit.org/show_bug.cgi?id=276515

Reviewed by Don Olmstead.

<https://commits.webkit.org/227764@main> added a workaround for zlib-ng.
zlib-ng fixed the problem. <https://github.com/zlib-ng/zlib-ng/pull/704>

* Source/ThirdParty/ANGLE/third_party/zlib/google/compression_utils_portable.cc:

Canonical link: https://commits.webkit.org/280930@main
